platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -98,63 +106,50 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpSolrClientFactory.class);
 
   @Override
-  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
-    String solrUrl =
-        StringUtils.defaultIfBlank(
-            AccessController.doPrivileged(
-                (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
-            getDefaultHttpsAddress());
-    final String coreUrl = solrUrl + "/" + core;
-    final String solrDataDir =
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty(SOLR_DATA_DIR));
-
-    if (solrDataDir != null) {
-      ConfigurationStore.getInstance().setDataDirectoryPath(solrDataDir);
+  public org.codice.solr.client.solrj.SolrClient newClient(String coreName) {
+    Args.notEmpty(coreName, "Cannot create Solr client. Missing core name.");
+    if (getSolrDir() != null) {
```

Comment:
❗️ Please saved the value info a local variable to avoid calling `getSolrDir()` twice which ends up doing a privileged action each time.